### ISSUE TITLE
[a11y]: Tearsheet without interactive elements not scrollable

### ISSUE THREAD

**Issue by miriam-g** [carbon-design-system/ibm-products#7521]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.64.0

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Accessiblity Checker- latest deployment

### Description

We have a tearsheet with different tabs and tabpanels. One panel has non-interactive content that is that large that it has a scrollbar. Scrolling doesn't work with the keyboard.

https://github.com/user-attachments/assets/72dd78ac-1683-4e1b-bc17-024480d4a891

I see this violation when using the Accessibility Checker:

![Image](https://github.com/user-attachments/assets/b1f2bc89-f2db-46be-b9e5-6d40153e3ebc)

The tabindex is set hard-coded, this leads to this issue:
https://github.com/carbon-design-system/ibm-products/blob/ce6686c6e0f48a233ac60ef709aded199921042d/packages/ibm-products/src/components/Tearsheet/TearsheetShell.tsx#L541

Same issue on Firefox and Chrome (v136).

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/iframe.html?globals=viewport%3Abasic&id=ibm-products-components-tearsheet--with-navigation&viewMode=story

### Steps to reproduce

To reproduce the issue, use this link: [carbon tearsheet with tabs](https://carbon-for-ibm-products.netlify.app/iframe.html?globals=viewport%3Abasic&id=ibm-products-components-tearsheet--with-navigation&viewMode=story) change the content on one tab that it doesn't fit on the page and use the keyboard to navigate through it.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by amal-k-joy** [comment]
`<section class="c4p--tearsheet__content">` has overflow auto , but the `tabindex` is hardcoded to `-1`

**Comment by miriam-g** [comment]
@amal-k-joy we can **not** ignore this because the content is not scrollable via keyboard (see my video) and hence it is not accessible/not readable.

**Comment by wkeese** [comment]
> `<section class="c4p--tearsheet__content">` has overflow auto , but the `tabindex` is hardcoded to `-1`. How ever we can ignore this error after correcting `tabindex` since its identified as [false positive](https://ibm-studios.slack.com/archives/C036P1CTN/p1744116605709549).

Exactly.

I made a PR at #7536.

I also made a simpler test case at https://stackblitz.com/edit/github-ykbrpwcg-f4hnrsfw?file=src%2FExample%2FExample.jsx.
